packages/edit-site/src/components/editor/global-styles-provider.js

```diff
@@ -191,14 +203,15 @@ export default function GlobalStylesProvider( { children, baseStyles } ) {
 				setContent( JSON.stringify( newContent ) );
 			},
 			getStyle: ( context, propertyName, origin = 'merged' ) => {
-				const styleOrigin =
-					'user' === origin ? userStyles : mergedStyles;
+				const styleOrigin = getStyleOrigin( origin );
 
 				const value = get(
 					styleOrigin?.styles?.[ context ],
 					STYLE_PROPERTY[ propertyName ].value
 				);
-				return getValueFromVariable( mergedStyles, context, value );
+
+				const styles = 'theme' === origin ? themeStyles : mergedStyles;
+				return getValueFromVariable( styles, context, value );
```

Comment:
I understand why we use both `styleOrigin` and `styles`. However, it's complex and I'm wondering how we can rewrite it so it's more straightforward for the reader. How about?

```js
getStyle: ( context, propertyName, origin = 'merged' ) => {
  if ( origin === 'theme' ) {
    const value = get( themeStyles, path );
    return getValueFromVariable( themeStyles, context, value );
  }

  if ( origin === 'user' ) {
    const value = get( userStyles, path );
    // We still need to use merged styles here because
    // the presets used to resolve user variable may be defined
    // a layer down (core, theme, or user).
    return getValueFromVariable( mergedStyles, context, value );
  }

  const value = get( mergedStyles, path );
  return getValueFromVariable( mergedStyles, context, value );
},
```

Reply:
Good idea 👍 

I've updated getStyle to the suggested structure. Thanks.